# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BytecodeAnalysis.cpp
# Commit: eeaeb40f8b86
# Full Hash: eeaeb40f8b869121eff65b1c2d11af423c9a9ecc
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2022-03-09 09:44:44
# Regressor Bug: 885514
# File Overlap Count: 1
# Description:
#   Bug 885514: Simplify JSOp::Retsub r=jandem
#   
#   Currently, JSOp::Retsub takes two arguments, and uses one to decide whether to throw the second or use it as a resume index. This patch splits out the exception-handling logic into explicit bytecode, leaving JSOp::Retsub as a simple indirect jump.
#   
#   To make the generated bytecode work out nicely, I've swapped the order of the arguments: now `throwing` is on top, and the exception/resumeindex argument is underneath it.
# ==============================================================================

diff -r 1cb0b352b024 -r eeaeb40f8b86 js/src/jit/BytecodeAnalysis.cpp
--- a/js/src/jit/BytecodeAnalysis.cpp	Tue Mar 08 21:10:22 2022 +0000
+++ b/js/src/jit/BytecodeAnalysis.cpp	Tue Mar 08 21:10:22 2022 +0000
@@ -143,10 +143,9 @@
       case JSOp::ResumeIndex: {
         // ResumeIndex is used to push a return address for a finally block. If
         // this op is reachable, then so is that return address (with a smaller
-        // stack depth because the resume index and `throwing` will have been
-        // popped.
+        // stack depth because the resume index will have been popped.
         uint32_t resumeOffset = script_->resumeOffsets()[(it.getResumeIndex())];
-        infos_[resumeOffset].init(stackDepth - 2);
+        infos_[resumeOffset].init(stackDepth - 1);
         infos_[resumeOffset].setJumpTarget(normallyReachable);
         break;
       }